emscripten_browser_clipboard.h

```diff
@@ -50,7 +50,30 @@ EM_JS_INLINE(void, copy_js, (copy_handler callback, void *callback_data), {
 
 EM_JS_INLINE(void, copy_async_js, (char const *content_ptr), {
   /// Attempt to copy the provided text asynchronously
-  navigator.clipboard.writeText(UTF8ToString(content_ptr));
+  if (navigator.clipboard && window.isSecureContext) {
+    return navigator.clipboard.writeText(UTF8ToString(content_ptr))
+           .then(()  => {})
+           .catch(() => {alert("something went wrong");});
+  } else {
+  /// Fallback method using document.execCommand('copy'), 
+  /// this fallback method is not guaranteed to work in all browsers since
+  /// document.execCommand('copy') is deprecated.  
```

Comment:
```suggestion
    // Fallback method using document.execCommand('copy'), 
    // this fallback method is not guaranteed to work in all browsers since
    // document.execCommand('copy') is deprecated.  
```